modules/artifact/src/test/java/org/mule/runtime/module/artifact/api/classloader/FilteringArtifactClassLoaderTestCase.java

```diff
@@ -133,6 +137,61 @@ public void loadsExportedResource() throws ClassNotFoundException, IOException {
     assertThat(resource, equalTo(expectedResource));
   }
 
+  @Test
+  public void getPackageExported() throws ClassNotFoundException, IOException {
+    TestClassLoader classLoader = new TestClassLoader(null);
+    classLoader.addClass(CLASS_NAME, this.getClass());
+
+    when(filter.exportsPackage(CLASS_PACKAGE_NAME)).thenReturn(true);
+    when(artifactClassLoader.getClassLoader()).thenReturn(classLoader);
+
+    filteringArtifactClassLoader = doCreateClassLoader(emptyList());
+    Package aPackage = filteringArtifactClassLoader.getPackage(CLASS_PACKAGE_NAME);
+    assertThat(aPackage.getName(), equalTo(CLASS_PACKAGE_NAME));
+  }
+
+  @Test
+  public void getPackageNoExported() throws ClassNotFoundException, IOException {
+    TestClassLoader classLoader = new TestClassLoader(null);
+    classLoader.addClass(CLASS_NAME, this.getClass());
+
+    when(filter.exportsPackage(CLASS_PACKAGE_NAME)).thenReturn(false);
+    when(artifactClassLoader.getClassLoader()).thenReturn(classLoader);
+
+    filteringArtifactClassLoader = doCreateClassLoader(emptyList());
+    Package aPackage = filteringArtifactClassLoader.getPackage(CLASS_PACKAGE_NAME);
+    assertThat(aPackage, is(nullValue()));
+  }
+
+  @Test
+  public void getPackagesExported() throws ClassNotFoundException, IOException {
+    TestClassLoader classLoader = new TestClassLoader(null);
+    classLoader.addClass(CLASS_NAME, this.getClass());
+    classLoader.addClass(ANOTHER_CLASS_NAME, this.getClass());
+
+    when(filter.exportsPackage(CLASS_PACKAGE_NAME)).thenReturn(true);
+    when(filter.exportsPackage(ANOTHER_CLASS_PACKAGE_NAME)).thenReturn(false);
+    when(artifactClassLoader.getClassLoader()).thenReturn(classLoader);
+
+    filteringArtifactClassLoader = doCreateClassLoader(emptyList());
+    Package[] packageList = filteringArtifactClassLoader.getPackages();
+    assertThat(packageList.length, is(1));
+    assertThat(packageList[0].getName(), equalTo(CLASS_PACKAGE_NAME));
+  }
+
+  @Test
+  public void getPackagesNoExported() throws ClassNotFoundException, IOException {
```

Comment:
This test is covered on getPackagesExported test with the line when(filter.exportsPackage(ANOTHER_CLASS_PACKAGE_NAME)).thenReturn(false);